### ISSUE TITLE
Can't build Godot with Mono support on Linux/X11

### ISSUE THREAD

**Issue by zatherz** [godotengine/godot#13285]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Arch Linux, latest git master Godot

**Issue description:**
Same error with `llvm=yes` and without.
```
$ scons module_mono_enabled=yes platform=x11 llvm=yes
scons: Reading SConscript files ...
Enabling ALSA
Enabling PulseAudio
YASM is necessary for WebM SIMD optimizations.
WebM SIMD optimizations are disabled. Check if your CPU architecture, CPU bits or platform are supported!
Checking for C header file mntent.h... (cached) yes
scons: done reading SConscript files.
scons: Building targets ...
[ 25%] Compiling ==> modules/mono/editor/bindings_generator.cpp
modules/mono/editor/bindings_generator.cpp: In member function 'Error BindingsGenerator::generate_cs_core_project(const String&, bool)':
modules/mono/editor/bindings_generator.cpp:373:35: error: invalid conversion from 'const unsigned char*' to 'uint8_t* {aka unsigned char*}' [-fpermissive]
   Compression::decompress(data.ptr(), file_data.uncompressed_size, file_data.data, file_data.compressed_size, Compression::MODE_DEFLATE);
                           ~~~~~~~~^~
In file included from modules/mono/editor/bindings_generator.cpp:36:0:
core/io/compression.h:53:13: note:   initializing argument 1 of 'static int Compression::decompress(uint8_t*, int, const uint8_t*, int, Compression::Mode)'
  static int decompress(uint8_t *p_dst, int p_dst_max_size, const uint8_t *p_src, int p_src_size, Mode p_mode = MODE_ZSTD);
             ^~~~~~~~~~
In file included from core/dvector.h:33:0,
                 from core/variant.h:41,
                 from core/method_ptrcall.h:35,
                 from core/method_bind.h:34,
                 from core/class_db.h:33,
                 from modules/mono/editor/bindings_generator.h:33,
                 from modules/mono/editor/bindings_generator.cpp:30:
modules/mono/editor/bindings_generator.cpp:385:20: error: invalid conversion from 'const void*' to 'void*' [-fpermissive]
    copymem(data.ptr(), reinterpret_cast<const uint8_t *>(data_utf8.get_data()), data_utf8.length());
            ~~~~~~~~^~
core/os/copymem.h:43:41: note: in definition of macro 'copymem'
 #define copymem(to, from, count) memcpy(to, from, count)
                                         ^~
In file included from core/os/copymem.h:41:0,
                 from core/dvector.h:33,
                 from core/variant.h:41,
                 from core/method_ptrcall.h:35,
                 from core/method_bind.h:34,
                 from core/class_db.h:33,
                 from modules/mono/editor/bindings_generator.h:33,
                 from modules/mono/editor/bindings_generator.cpp:30:
/usr/include/string.h:42:14: note:   initializing argument 1 of 'void* memcpy(void*, const void*, size_t)'
 extern void *memcpy (void *__restrict __dest, const void *__restrict __src,
              ^~~~~~
scons: *** [modules/mono/editor/bindings_generator.x11.tools.64.mono.o] Error 1
scons: building terminated because of errors.
```

**Comment by vnen** [comment]
Likely related to changes from bc2e8d99e5ae0dbd69e712cc71da3033f5f30139.

**Comment by akien-mga** [comment]
Your git clone must have been a few commits behind, this was fixed already in aa5f7e0ff.
